jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -178,9 +185,20 @@ public void doWrite(Server server, Query query, Iterable<Result> results) throws
 
 			HashMap<String, Object> filteredValues = newHashMap();
 			Object value = result.getValue();
+
+			ImmutableList<String> typeNamesParam = this.typeNames;
+			// if typeNamesAsTag, we don't concat typeName in values.
+			if (typeNamesAsTags) {
+				typeNamesParam = ImmutableList.<String>of();
+			}
+
+			String key = KeyUtils.getPrefixedKeyString(query, result, typeNames);
 			if (isValidNumber(value)) {
-				String key = KeyUtils.getPrefixedKeyString(query, result, typeNames);
 				filteredValues.put(key, value);
+			}else {
+				if (segregateStringValues) {
```

Comment:
If `value` is not a valid number, it does not mean it is a `String`. It can still be any kind of object. So there is at least some naming issue with `segregateStringValues`, possibly a conceptual issue as well.

Reply:
We have metrics that change from time to time, and influx does not like that, and if a metric is sent at the beginning in boolean, it will not be accepted anymore if it corrects itself and sends its numerical data.

putting everything in String is for simplicity, and can have a history of values sent in error (String and boolean)

Reply:
is it Ok if I rename to segregateNoNumValues ?

Reply:
For the boolean case, there is a [BooleanAsNumberValueTransformer](https://github.com/jmxtrans/jmxtrans/blob/master/jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/results/BooleanAsNumberValueTransformer.java) that might need to be added to the InfluxDB writer.

For the more generic case of segregating non numeric values, yes, your solution seems to make sense. Please rename the parameter as `segregateNonNumericValues` and add a parameter for the suffix. Also, a few lines of javadoc on the parameter explaining the use case would be welcomed!

Reply:
My understanding is that @yassineLajmi has mbeans that for the same attribute will sometime return a boolean and sometime an integer. The change might happen with a new versio of the application under monitoring, or just with a badly written mbean.